pkg/cmd/util/env.go

```diff
@@ -93,6 +100,8 @@ func ParseEnv(spec []string, defaultReader io.Reader) ([]kapi.EnvVar, []string,
 	var remove []string
 	for _, envSpec := range spec {
 		switch {
+		case !IsValidEnvironmentArgument(envSpec) && !strings.HasSuffix(envSpec, "-"):
+			return nil, nil, fmt.Errorf("environment variables must be of the form key=value and match the pattern: %v\n", validEnvArgRegex)
```

Comment:
"environment variables must be of the form key=value and key can only contain letters, digits and underscore" or something similar